packages/arb-token-bridge-ui/src/components/common/NetworkSelectionContainer.tsx

```diff
@@ -0,0 +1,330 @@
+import { useNetworks } from '../../hooks/useNetworks'
+
+type NetworkInfo = {
+  chainId: number
+  type: 'core' | 'orbit'
+}
+
+const chainGroupInfo = {
+  core: {
+    name: 'CORE CHAINS',
+    description: 'Chains managed directly by Ethereum or Arbitrum'
+  },
+  orbit: {
+    name: 'ORBIT CHAINS',
+    description: 'Independent chains supported by Arbitrum technology'
+  }
+}
+
+function NetworkRow({
+  networkInfo,
+  style,
+  onClick,
+  close
+}: {
+  networkInfo: NetworkInfo
+  style: CSSProperties
+  onClick: (value: Chain) => void
+  close: (focusableElement?: HTMLElement) => void
+}) {
+  const { chainId } = networkInfo
+  const { network } = getBridgeUiConfigForChain(chainId)
+  const chain = getWagmiChain(chainId)
+
+  const handleClick = () => {
+    onClick(chain)
+    close() // close the popover after option-click
+  }
+
+  return (
+    <button
+      onClick={handleClick}
+      onKeyDown={e => {
+        if (e.key === 'Enter') {
+          handleClick()
+        }
+      }}
+      key={chainId}
+      style={style}
+      type="button"
+      aria-label={`Switch to ${network.name}`}
+      className={twMerge(
+        'flex h-[90px] w-full items-center gap-4 px-6 py-2 text-lg hover:bg-black/10'
+      )}
+    >
+      <span className="flex h-6 w-6 shrink-0 items-center justify-center lg:h-6 lg:w-6">
+        <Image
+          src={network.logo}
+          alt={`${network.name} logo`}
+          className="h-full w-auto"
+          width={24}
+          height={24}
+        />
+      </span>
+      <div className={twMerge('flex flex-col items-start gap-1')}>
+        <span className="truncate leading-none">{network.name}</span>
+        {chainInfo[chainId] && (
+          <>
+            <p className="whitespace-pre-wrap text-left text-xs leading-[1.15]">
+              {chainInfo[chainId]!.description}
+            </p>
+            <p className="text-[10px] leading-none">
+              {chainInfo[chainId]!.chainType} Chain,{' '}
+              {chainInfo[chainId]!.nativeCurrency} is the native gas token
+            </p>
+          </>
+        )}
+      </div>
+    </button>
+  )
+}
+
+function AddCustomOrbitChainButton() {
+  const [, setQueryParams] = useArbQueryParams()
+  const { isTestnetMode } = useIsTestnetMode()
+
+  if (!isTestnetMode) {
+    return null
+  }
+
+  return (
+    <button
+      className="text-sm underline"
+      onClick={() => setQueryParams({ settingsOpen: true })}
+    >
+      <span>Add Custom Orbit Chain</span>
+    </button>
+  )
+}
+
+function NetworksPanel({
+  networks,
+  onNetworkRowClick,
+  close
+}: {
+  networks: NetworkInfo[]
+  onNetworkRowClick: (value: Chain) => void
+  close: (focusableElement?: HTMLElement) => void
+}) {
+  const [errorMessage, setErrorMessage] = useState('')
+  const [networkSearched, setNetworkSearched] = useState('')
+  const [debouncedNetworkSearched, setDebouncedNetworkSearched] = useState('')
+
+  useDebounce(
```

Comment:
Do we need to debounce this? We don't make any expensive searches here or network calls.

Reply:
otherwise `networksToShowWithChainTypeInfo` would be run every time user types though

i think debouncing is a good practice